### ISSUE TITLE
Is HTTP BASIC authentication sufficient?

### ISSUE THREAD

**Issue by franciscave** [bic-org-uk/bic-lcf#285]
At the PlugFest on 11 May 2021 there was discussion about whether LCF should document specific alternatives to HTTP BASIC authentication. At least one PlugFest participant was expressing the view that HTTP BASIC was not considered adequate by some implementers, and that OAuth should be reconsidered as a preferred alternative to HTTP BASIC.

This topic has been discussed in the past, but not for a while and it is probably worth reviewing what is now appropriate in this area.

**Comment by mdovey** [comment]
Agreed at technical panel on the 3/11/2021, to investigate alternative authentications after the next release, but to make it clear int the documentation that other authentication mechanisms may be used.

**Comment by franciscave** [comment]
I propose the following changes to the REST web service specification, to indicate that other methods may be used for authentication. I am assuming that bearer token authentication (as implemented by JSON Web Token) is specifically relevant to *user*, i.e. Patron, authentication, but that this or other authentication methods may also be applied to terminal authentication. I therefore propose adding text under both terminal application authentication and patron authentication, but only mentioning bearer token authentication explicitly under patron authentication.

### Implementation notes

#### 1. Terminal Application Authentication *(updated for vx.x.0)*

...

**D** Other standard terminal authentication methods.

#### 2. Patron Authentication *(updated for vx.x.0)*

In addition to terminal application authentication, an LMS will frequently require that the patron user of the terminal application be themselves authenticated. The LCF elements Q00D01.2 and Q00D02.2 should be used for this purpose. A RESTful web service implementation of LCF may use any of the following methods for authentication of patrons, but method B is *not* recommended, being the least secure: 

...

**C** JSON Web Token or other standard bearer token method.

**Comment by franciscave** [comment]
Following discussion in the Technical Panel meeting today, I have revised my proposed changes to the REST web service specification, to specify the same additional choice of authentication method for both terminal and patron authentication. The following changes are implemented in pull request #291.

### Implementation notes

#### 1. Terminal Application Authentication *(updated for vx.x.0)*

...

**D** Other standard terminal authentication methods, such as bearer token authentication.

#### 2. Patron Authentication *(updated for vx.x.0)*

...

**C** Other standard terminal authentication methods, such as bearer token authentication.
